src/index.ts

```diff
@@ -74,57 +138,76 @@ import { IconAddBorder, IconStretch, IconAddBackground, IconPicture } from '@cod
  *                           also can contain any additional data that will be saved and passed back
  * @property {string} file.url - [Required] image source URL
  */
-export default class ImageTool {
+interface UploadResponseFormat {
   /**
-   * Notify core that read-only mode is supported
-   *
-   * @returns {boolean}
+   * success - 1 for successful uploading, 0 for failure 
    */
-  static get isReadOnlySupported() {
-    return true;
-  }
-
+  success: number;
   /**
-   * Get Tool toolbox settings
-   * icon - Tool icon's SVG
-   * title - title to show in toolbox
-   *
-   * @returns {{icon: string, title: string}}
+   * Object with file data.
+   *             'url' is required,
+   *             also can contain any additional data that will be saved and passed back
    */
-  static get toolbox() {
-    return {
-      icon: IconPicture,
-      title: 'Image',
-    };
-  }
+  file: {
+    /**
+     * The URL of the uploaded image.
+     */
+    url: string;
+  };
+}
 
+interface BlockToolConstructorOptions {
   /**
-   * Available image tools
-   *
-   * @returns {Array}
+   * Previously saved data as ImageTool data.
    */
-  static get tunes() {
-    return [
-      {
-        name: 'withBorder',
-        icon: IconAddBorder,
-        title: 'With border',
-        toggle: true,
-      },
-      {
-        name: 'stretched',
-        icon: IconStretch,
-        title: 'Stretch image',
-        toggle: true,
-      },
-      {
-        name: 'withBackground',
-        icon: IconAddBackground,
-        title: 'With background',
-        toggle: true,
-      },
-    ];
-  }
+  data: ImageToolData;
+  /**
+   * User config for Tool.
+   */
+  config: ImageToolConfig;
+  /**
+   * Editor.js API.
+   */
+  api: API;
+  /**
+   * Flag indicating read-only mode.
+   */
+  readOnly: boolean;
+  /**
+   * Current Block API.
+   */
+  block: any;
+}
+
+export default class ImageTool implements BaseTool{
```

Comment:
`BaseTool` is the base class for all tools, including inline tools and block tunes. You need to implement `BlockTool` interface 

```suggestion
export default class ImageTool implements BlockTool {
```